### PR TITLE
Add configuration flag to enable next protocol version

### PR DESCRIPTION
**Pull request by jonjove**
# Description
Adds the configure flag `--enable-next-protocol-version-unsafe-for-production` which makes it possible to develop protocol changes further in advance of a protocol release

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -337,6 +337,12 @@ AC_ARG_ENABLE(spdlog,
         [Disable spdlog]))
 AM_CONDITIONAL(USE_SPDLOG, [test x$enable_spdlog != xno])
 
+AC_ARG_ENABLE(next-protocol-version-unsafe-for-production,
```
Comment (MonsieurNicolas) [inline configure.ac:340]: should this be enabled in the ci script?

Comment (jonjove) [reply configure.ac:340]: Yes it should. I missed that. I will update shortly.

Comment (jonjove) [reply configure.ac:340]: Fixed.

**Comment by MonsieurNicolas** [comment]
r+ 6eb43f6ee31f3ed78eebcb2f7183ef939b83b12a

**Pull request merged**